HLTrigger/Configuration/python/customizeHLTforPatatrack.py

```diff
@@ -286,112 +316,151 @@ def customisePixelTrackReconstruction(process):
     hasHLTPixelVertexReco = 'HLTRecopixelvertexingSequence' in process.__dict__
 
     # FIXME replace the Sequences with empty ones to avoid expanding them during the (re)definition of Modules and EDAliases
-
     process.HLTRecoPixelTracksSequence = cms.Sequence()
     if hasHLTPixelVertexReco:
         process.HLTRecopixelvertexingSequence = cms.Sequence()
 
 
     # Modules and EDAliases
-
     # referenced in process.HLTRecoPixelTracksTask
 
-    # build pixel ntuplets and pixel tracks in SoA format on gpu
-    from RecoPixelVertexing.PixelTriplets.pixelTracksCUDA_cfi import pixelTracksCUDA as _pixelTracksCUDA
-    process.hltPixelTracksCUDA = _pixelTracksCUDA.clone(
-        idealConditions = False,
-        pixelRecHitSrc = "hltSiPixelRecHitsCUDA",
-        onGPU = True
-    )
-    # use quality cuts tuned for Run 2 ideal conditions for all Run 3 workflows
-    run3_common.toModify(process.hltPixelTracksCUDA, idealConditions = True)
-
-    # SwitchProducer providing the pixel tracks in SoA format on cpu
-    from RecoPixelVertexing.PixelTrackFitting.pixelTracksSoA_cfi import pixelTracksSoA as _pixelTracksSoA
-    process.hltPixelTracksSoA = SwitchProducerCUDA(
-        # build pixel ntuplets and pixel tracks in SoA format on cpu
-        cpu = _pixelTracksCUDA.clone(
+    # SwitchProducer: hltPixelTracksSoA
+    if not ('hltPixelTracksSoA' in process.__dict__ and isinstance(process.hltPixelTracksSoA, SwitchProducerCUDA)):
+        # build pixel ntuplets and pixel tracks in SoA format on gpu
+        from RecoPixelVertexing.PixelTriplets.pixelTracksCUDA_cfi import pixelTracksCUDA as _pixelTracksCUDA
+        process.hltPixelTracksCUDA = _pixelTracksCUDA.clone(
+            pixelRecHitSrc = "hltSiPixelRecHitsCUDA",
+            onGPU = True,
             idealConditions = False,
-            pixelRecHitSrc = "hltSiPixelRecHitSoA",
-            onGPU = False
-        ),
-        # transfer the pixel tracks in SoA format to the host
-        cuda = _pixelTracksSoA.clone(
+        )
+        # use quality cuts tuned for Run-2 ideal conditions for all Run-3 workflows
+        run3_common.toModify(process.hltPixelTracksCUDA, idealConditions = True)
+
+        process.hltPixelTracksSoA = SwitchProducerCUDA(
+            # build pixel ntuplets and pixel tracks in SoA format on cpu
+            cpu = process.hltPixelTracksCUDA.clone(
+                pixelRecHitSrc = "hltSiPixelRecHitSoA",
+                onGPU = False,
+            )
+        )
+
+    elif hasattr(process.hltPixelTracksSoA, 'cpu'):
+        # if cpu branch of SwitchProducerCUDA exists, take hltPixelTracksCUDA (gpu)
+        # from hltPixelTracksSoA.cpu (cpu) to enforce same configuration parameters
+        process.hltPixelTracksCUDA = process.hltPixelTracksSoA.cpu.clone(
+            pixelRecHitSrc = "hltSiPixelRecHitsCUDA",
+            onGPU = True
+        )
```

Comment:
This relates to the treatment of the value of `process.hltPixelTracksSoA.idealConditions`. If `process.hltPixelTracksSoA.cuda` exists, it is not modified, and the GPU module (`hltPixelTracksCUDA`) is configured based on `process.hltPixelTracksSoA.cuda`.